### PR TITLE
[FIX] Issue #342 [12.0][BUG] agreement_legal: new agreement button (template form)

### LINKED ISSUE #342: [12.0][BUG] agreement_legal: new agreement button (template form)
1. Create a template
2. add at least 2 sections
3. click on button "new agreement" generate the following error:

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/models.py", line 4698, in ensure_one
    _id, = self._ids
ValueError: too many values to unpack (expected 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 654, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 312, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/.repo_requirements/odoo/odoo/http.py", line 696, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 344, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 337, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 939, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 517, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/OCB-12.0/addons/web/controllers/main.py", line 966, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/odoo/OCB-12.0/addons/web/controllers/main.py", line 954, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 749, in call_kw
    return _call_kw_multi(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 736, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/odoo/build/OCA/contract/agreement_legal/models/agreement.py", line 353, in create_new_agreement
    res.sections_ids.clauses_ids.write({'agreement_id': res.id})
  File "/.repo_requirements/odoo/odoo/fields.py", line 979, in __get__
    record.ensure_one()
  File "/.repo_requirements/odoo/odoo/models.py", line 4701, in ensure_one
    raise ValueError("Expected singleton: %s" % self)
ValueError: Expected singleton: agreement.section(3, 4)
```

### PR DESCRIPTION
**Pull request by ygol**
fix issue #342 

1. Create a template
2. add at least 2 sections
3. click on button "new agreement" generate the following error:

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/models.py", line 4698, in ensure_one
    _id, = self._ids
ValueError: too many values to unpack (expected 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 654, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 312, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/.repo_requirements/odoo/odoo/http.py", line 696, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 344, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 337, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 939, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 517, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/OCB-12.0/addons/web/controllers/main.py", line 966, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/odoo/OCB-12.0/addons/web/controllers/main.py", line 954, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 749, in call_kw
    return _call_kw_multi(method, model, args, kwargs)
  File "/.repo_requirements/odoo/odoo/api.py", line 736, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/odoo/build/OCA/contract/agreement_legal/models/agreement.py", line 353, in create_new_agreement
    res.sections_ids.clauses_ids.write({'agreement_id': res.id})
  File "/.repo_requirements/odoo/odoo/fields.py", line 979, in __get__
    record.ensure_one()
  File "/.repo_requirements/odoo/odoo/models.py", line 4701, in ensure_one
    raise ValueError("Expected singleton: %s" % self)
ValueError: Expected singleton: agreement.section(3, 4)
```

### CODE REVIEW

File: agreement_legal/models/agreement.py [hunk 1]
```diff
@@ -350,7 +350,7 @@ def create_new_agreement(self):
             "stage_id": self.env.ref("agreement_legal.agreement_stage_new").id,
         }
         res = self.copy(default=default_vals)
-        res.sections_ids.clauses_ids.write({'agreement_id': res.id})
+        res.sections_ids.write({'agreement_id': res.id})
```
Comment (kongrattapong) [inline agreement_legal/models/agreement.py]:
```suggestion
        res.sections_ids.mapped('clauses_ids').write({'agreement_id': res.id})
```
@ygol I got bug form this line 
Step:
1.Create 2 Sections : Sections A and Sections B.
2.Create 3 Clauses to Sections B
3.Create New Agreement.
and I got this in agreement template
BUY-VOIP013  SIP-ISDN gateways - Odoo - Google Chrome_003

Comment (ygol) [reply agreement_legal/models/agreement.py]: @kongrattapong Thanks for reviewing. I'll check it ASAP

**Comment by rafaelbn** [comment]
Hi @ygol , I think it much more better to have in the same PR the tittle and all the information of the issue and not have duplicated and separated the info, the issue and the solution. This thinking in having reviewers.

**Comment by ygol** [comment]
> Hi @ygol , I think it much more better to have in the same PR the tittle and all the information of the issue and not have duplicated and separated the info, the issue and the solution. This thinking in having reviewers.

Hi @rafaelbn  tnx for feedback. 
Do you mean paste here (PR) - in the title - the issue's full title? + paste in the body, the issue's full detail?

**Comment by rafaelbn** [comment]
Hi @ygol , yes. Usually an issue is to report a problem that you cannot or you don't know how to solve. But if you are going to solve your self then when opening the PR with the fix put the same tittle and the detailed description :smile: and the report and the solution is together so will be easier to review and merge. And you will attract more reviewers!

File: agreement_legal/models/agreement.py (hunk 1 above)
Comment (ygol) [reply agreement_legal/models/agreement.py]: @kongrattapong thanks for catching & solving this bug. If everything looks fine please approve.

**Comment by ygol** [comment]
@max3903 would you please review this PR and approve? TIA

**Review by kongrattapong — APPROVED** [review]
Code reviewed :+1:

**Comment by ygol** [comment]
@max3903 would you please review and merge

**Comment by pedrobaeza** [comment]
@max3903 please confirm this.

**Review by max3903 — APPROVED** [review]

**Pull request merged**